cypress/support/step_definitions/dynamicCalls.js

```diff
@@ -348,6 +348,20 @@ Given(
  * Given 'third party app is launched' on 'auth' page
  */
 Given(/'(.+)' on '(.+)' page/, (validationObjectKey, page) => {
+  const requestMap = {
+    method: CONSTANTS.REQUEST_OVERRIDE_CALLS.GETAPPSTATE,
+    params: Cypress.env(CONSTANTS.CURRENT_APP_ID),
+  };
+
+  // Sending the request to the platform to retrieve the app state.
+  cy.sendMessagetoPlatforms(requestMap).then((response) => {
+    if (response === 'FOREGROUND') {
+      fireLog.info(`State validation successful: Current state is '${response}' as expected`);
```

Comment:
constants can be used here. what if foreground comes as 'foreground' ?

Reply:
Added .toUpperCase() to handle case sensitivity in the response. In Xumo, getting the response in uppercase ('FOREGROUND'). My doubt is whether we need to pass both lowercase and uppercase cases. Is that the expectation?

Reply:
why should you write FCS code based on what xumo returns ? 
It should pass both foreground and FOREGROUND. Converting to uppercase is also fine as it will handle both